File: back-kotlin/src/main/kotlin/me/nicolas/stravastats/domain/business/strava/Bike.kt
package me.nicolas.stravastats.domain.business.strava

import com.fasterxml.jackson.annotation.JsonProperty

data class Bike(
    @param:JsonProperty("distance")
    val distance: Int,
    @param:JsonProperty("id")
    val id: String,
    @param:JsonProperty("name")
    val name: String,
    @param:JsonProperty("nickname")
    val nickname: String?,
    @param:JsonProperty("retired")
    val retired: Boolean?,
    @param:JsonProperty("converted_distance")
    val convertedDistance: Double,
    @param:JsonProperty("primary")
    val primary: Boolean,
    @param:JsonProperty("resource_state")
    val resourceState: Int,
)

File: back-go/domain/strava/bike.go
package strava

type Bike struct {
	Distance          int     `json:"distance"`           // Distance covered by the bike, in meters
	Id                string  `json:"id"`                 // Unique identifier of the bike
	Name              string  `json:"name"`               // Name of the bike
	Nickname          *string `json:"nickname,omitempty"` // Optional nickname for the bike
	Retired           *bool   `json:"retired,omitempty"`  // Indicates if the bike is retired (optional)
	ConvertedDistance float64 `json:"converted_distance"` // Distance converted to another unit (e.g., kilometers)
	Primary           bool    `json:"primary"`            // True if this is the primary bike
	ResourceState     int     `json:"resource_state"`     // Level of detail about the bike resource
}
